pom.xml

```diff
@@ -0,0 +1,216 @@
+</dependencies>
+
+<build>
+  <plugins>
+    <plugin>
+      <groupId>org.apache.maven.plugins</groupId>
+      <artifactId>maven-compiler-plugin</artifactId>
+      <version>3.6.1</version>
+    </plugin>
+    <plugin>
+      <groupId>org.apache.maven.plugins</groupId>
+      <artifactId>maven-surefire-plugin</artifactId>
+      <version>3.0.0-M4</version>
+      <configuration>
+        <properties>
+          <configurationParameters>
+            junit.jupiter.execution.parallel.enabled=true
+            junit.jupiter.execution.parallel.mode.default=same_thread
+            junit.jupiter.execution.parallel.mode.classes.default=concurrent
+          </configurationParameters>
+        </properties>
+      </configuration>
+    </plugin>
+    <plugin>
+      <groupId>com.github.spotbugs</groupId>
+      <artifactId>spotbugs-maven-plugin</artifactId>
+      <version>3.1.12.2</version>
+      <executions>
+        <execution>
+          <id>spotbugs</id>
+          <phase>verify</phase>
+          <goals>
+            <goal>check</goal>
+          </goals>
+        </execution>
+      </executions>
+    </plugin>
+    <plugin>
+      <groupId>org.jacoco</groupId>
+      <artifactId>jacoco-maven-plugin</artifactId>
+      <version>0.8.5</version>
+      <executions>
+        <execution>
+          <goals>
+            <goal>prepare-agent</goal>
+          </goals>
+        </execution>
+        <execution>
+          <id>report</id>
+          <phase>verify</phase>
+          <goals>
+            <goal>report</goal>
+          </goals>
+        </execution>
+      </executions>
+    </plugin>
+    <plugin>
+      <groupId>com.diffplug.spotless</groupId>
+      <artifactId>spotless-maven-plugin</artifactId>
+      <version>1.27.0</version>
+      <executions>
+        <execution>
+          <id>spotless</id>
+          <phase>verify</phase>
+          <goals>
+            <goal>check</goal>
+          </goals>
+        </execution>
+      </executions>
+      <configuration>
+        <java>
+          <googleJavaFormat>
+            <version>1.7</version>
+            <style>AOSP</style>
+          </googleJavaFormat>
+          <trimTrailingWhitespace/>
+          <endWithNewline/>
+          <importOrder>
+            <order>java,javax,org.openjdk.jmc,com.redhat.rhjmc.containerjfr,</order>
+          </importOrder>
+          <removeUnusedImports/>
+        </java>
+      </configuration>
+    </plugin>
+    <plugin>
+      <groupId>org.apache.maven.plugins</groupId>
+      <artifactId>maven-site-plugin</artifactId>
+      <version>3.8.2</version>
+    </plugin>
+    <plugin>
+      <groupId>org.apache.maven.plugins</groupId>
+      <artifactId>maven-project-info-reports-plugin</artifactId>
+      <version>3.0.0</version>
+    </plugin>
+  </plugins>
+</build>
+
+<reporting>
+  <plugins>
+    <plugin>
+      <groupId>com.github.spotbugs</groupId>
+      <artifactId>spotbugs-maven-plugin</artifactId>
+      <version>3.1.12.2</version>
```

Comment:
Would it be worth defining all the plugin versions in the `properties` section? I think this one is duplicated in the file already.

Reply:
Yea, that's a good idea. Any idea or preference on how these properties should be named?

Reply:
How about:
artifactId.version with instances of '-' replaced by '.'

So:
spotbugs.maven.plugin.version

Hmm on further thought, for version definitions of plugins, you only need one instance and it could be all placed in a pluginManagement section. Something like:
https://github.com/jiekang/thermostat-web-gateway/blob/master/pom.xml

Thoughts?

Reply:
Er yeah, you described it perfectly. It's a bit verbose though and makes more sense when you have multiple maven submodules using things. Either solution is okay with me, whichever you prefer.